src/SAML2/XML/samlp/StatusCode.php

```diff
@@ -97,6 +97,9 @@ public function setSubCodes(?array $subCodes): void
      */
     public static function fromXML(DOMElement $xml): object
     {
+        Assert::same($xml->localName, 'StatusCode');
+        Assert::same($xml->namespaceURI, Constants::NS_SAMLP);
+
```

Comment:
I can't add a comment on the proper line, so I'll add it here. Line 147:

```php
        if (!empty($this->subCodes)) {
            foreach ($this->subCodes as $subCode) {
                $subCode->toXML($e);
            }
        }
```

The outer if can be removed I think, since if `$this->subCodes` is empty, the loop will not iterate.

Reply:
It's optional, so it could be null..

Reply:
My bad, didn't notice it could be null!